packages/dd-trace/src/startup-log.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const mainLogger = require('./log')
+const path = require('path')
+const { inspect } = require('util')
+const tracerVersion = require('../lib/version')
+
+let os
+try {
+  os = require('os')
+} catch (e) {
+  // in browser
+}
+
+const logger = Object.create(mainLogger)
+logger._enabled = process.env.DD_TRACE_STARTUP_LOGS !== '0'
+
+let config
+let plugins = []
+let samplingRules = []
+
+let alreadyRan = false
+
+function getIntegrations () {
+  const integrations = []
+  for (const plugin of plugins) {
+    if (plugin.versions) {
+      try {
+        const version = require(path.join(plugin.name, 'package.json')).version
+        integrations.push(`${plugin.name}@${version}`)
+      } catch (e) {
+        integrations.push(plugin.name)
+      }
+    } else {
+      integrations.push(plugin.name)
+    }
+  }
+  return Array.from(new Set(integrations))
+}
+
+function startupLog (agentError) {
+  if (!config) {
+    return
+  }
+
+  if (alreadyRan) {
+    return
+  }
+
+  alreadyRan = true
+
+  const url = config.url || `http://${config.hostname || 'localhost'}:${config.port}`
+
+  const out = {
+    [inspect.custom] () {
+      return String(this)
+    },
+    toString () {
+      return JSON.stringify(this)
+    }
+  }
+
+  out.date = new Date().toISOString()
+  out.os_name = os.type()
+  out.os_version = os.release()
+  out.version = tracerVersion
+  out.lang = 'nodejs'
+  out.lang_version = process.versions.node
+  out.env = config.env
+  out.enabled = config.enabled
+  out.scope_manager = config.scope
+  out.service = config.service
+  // out.enabled_cli // N/A
+  out.agent_url = url
+  if (agentError) {
+    out.agent_error = agentError.message
+  }
+  out.debug = !!config.debug
+  out.analytics_enabled = !!config.analytics
+  out.sample_rate = config.sampleRate
+  out.sampling_rules = samplingRules
+  // out.sampling_rules_error // N/A
```

Comment:
Should we add TODOs instead if these are supposed to be supported?

Reply:
They're only supposed to be supported if they actually make sense for the given language tracer. Other languages are implementing these if they make sense.

I put `N/A` on the ones I thought didn't apply to JS, so that it's clear that we're explicitly not supporting them. I'm hesitant to add `TODO`, since that would actually imply implementing the given feature in the tracer, which is outside the scope of a `TODO` in this file.

Maybe for now I can add a whole section saying "common fields not supported by the JS tracer" or something like that?

Also if we actually *can* provide some of this info, and I'm just mistaken, please let me know.

Reply:
I think it would make sense to have a section for this with a bit of an explanation, outside of the actual code. It will make it clearer that we may or may not support these in the future depending on the capabilities of the JS tracer.

Reply:
Latest rev has this in its own section.